src/runtime_src/core/pcie/tools/xbmgmt/cmd_config.cpp

```diff
@@ -225,24 +242,39 @@ static int show(int argc, char *argv[])
 }
 
 static void updateDevConf(std::shared_ptr<pcidev::pci_device>& dev,
-    std::string lvl)
+    std::string lvl, int config_type)
 {
     std::string errmsg;
-    dev->sysfs_put("icap", "sec_level", errmsg, lvl);
-    if (!errmsg.empty()) {
-        std::cout << "Failed to set security level for " << dev->sysfs_name
-            << std::endl;
-        std::cout << "See dmesg log for details" << std::endl;
+
+    switch(config_type) {
+    case CONFIG_SECURITY:
+        dev->sysfs_put("icap", "sec_level", errmsg, lvl);
+        if (!errmsg.empty()) {
+            std::cout << "Failed to set security level for " << dev->sysfs_name
+                << std::endl;
+            std::cout << "See dmesg log for details" << std::endl;
+        }
+        break;
+    case CONFIG_CLK_SCALING:
+        dev->sysfs_put("xmc", "scaling_enabled", errmsg, lvl);
+        if (!errmsg.empty()) {
+            std::cout << "Failed to update clk scaling status for " <<
+                dev->sysfs_name << std::endl;
+            std::cout << "See dmesg log for details" << std::endl;
+        }
+        break;
     }
 }
 
 static int device(int argc, char *argv[])
 {
     unsigned int index = UINT_MAX;
     std::string lvl;
+    int config_type = -1;
```

Comment:
There is no such enum value defined.   It is better to use a type enum for configType.

Reply:
config_type var used here to store actual enum configType values based on switch case condition.

Reply:
My point is that that 'int config_type' is misuse of an enum and that the code would be cleaner if it used a typed enum class.